src/main/java/org/apache/maven/plugins/enforcer/BanDuplicateClasses.java

```diff
@@ -217,73 +228,65 @@ else if ( file.isFile() && "jar".equals( o.getType() ) )
 
     }
 
-    private void checkAndAddName( Artifact artifact, String name, Map<String, Artifact> classNames,
-                                  Map<String, Set<Artifact>> duplicates, Collection<IgnorableDependency> ignores )
+    private void checkAndAddName( Artifact artifact, String pathToClassFile, Map<String, ClassesWithSameName> classesSeen,
+                                  Set<String> duplicateClasses, Collection<IgnorableDependency> ignores )
         throws EnforcerRuleException
     {
-        if ( !name.endsWith( ".class" ) )
+        if ( !pathToClassFile.endsWith( ".class" ) )
         {
             return;
         }
-        
+
         for ( IgnorableDependency c : ignores )
         {
-            if ( c.matchesArtifact( artifact ) && c.matches( name ) )
+            if ( c.matchesArtifact( artifact ) && c.matches( pathToClassFile ) )
             {
-                if( classNames.containsKey( name ) )
+                if ( classesSeen.containsKey( pathToClassFile ) )
                 {
-                    getLog().debug( "Ignoring excluded class " + name );
+                    getLog().debug( "Ignoring excluded class " + pathToClassFile );
                 }
                 return;
             }
         }
 
-        if ( classNames.containsKey( name ) )
+        ClassesWithSameName classesWithSameName = classesSeen.get( pathToClassFile );
+        boolean isFirstTimeSeeingThisClass = ( classesWithSameName == null );
+        ClassFile classFile = new ClassFile( pathToClassFile, artifact );
+
+        if ( isFirstTimeSeeingThisClass )
         {
-            Artifact dup = classNames.put( name, artifact );
-            if ( !( findAllDuplicates && duplicates.containsKey( name ) ) )
-            {
-                for ( IgnorableDependency c : ignores )
```

Comment:
It turns out we don't need this block since the earlier block where we loop over the `ignores` would already have returned when `c.matchesArtifact( artifact ) && c.matches(name)` is true.

In this PR that's lines 240 - 250.